src/hyperion/external_interaction/callbacks/robot_load/ispyb_callback.py

```diff
@@ -62,7 +62,12 @@ def activity_gated_event(self, doc: Event) -> Event | None:
                 self.action_id is not None
             ), "ISPyB Robot load callback event called unexpectedly"
             barcode = doc["data"]["robot-barcode"]
-            self.expeye.update_barcode(self.action_id, barcode)
+            oav_snapshot = doc["data"]["oav_snapshot_last_saved_path"]
```

Comment:
How do you know if you need dashes or hyphens in lines like `webcam_snapshot = doc["data"]["webcam-last_saved_path"]`?

Reply:
So when you have a device like:
```python
class MyDevice():
    self.signal = ...
my_device = MyDevice()
```
Then you do `yield from bps.read(my_device)` the `RE` produces a document with data like:
```
data: {
    "X": "value"
}
```
Where:
* In `ophyd` `_` is used as a separator so X is `my_device_signal`
* `ophyd_async` instead uses `-` so X is `my_device-signal`

If you tried to look in the dict for the wrong one you would fail with a `KeyError`. This is a frustrating breaking change between the two but it does make sense because `_` is a valid (and commonly used) part of a variable name it's impossible to know which was the device and signal in `my_device_signal` by just looking at the string. By using `-` instead you can work that out.